src/handle_provide_dynamic_descriptor.c

```diff
@@ -115,206 +84,78 @@ static bool handle_tuid_ext_code(const tlv_data_t *data, tlv_TUID_data_t *tlv_TU
     X(0x12, TUID_EXT_CODE,        handle_tuid_ext_code,        ENFORCE_UNIQUE_TAG)
 // clang-format on
 
-DEFINE_TLV_PARSER(TUID_TLV_TAGS, parse_dynamic_token_tuid)
-
-static bool handle_structure_type(const tlv_data_t *data, tlv_extracted_t *tlv_extracted) {
-    CX_ASSERT(cx_hash_update((cx_hash_t *) &tlv_extracted->hash_ctx, data->raw, data->raw_size));
-    return get_uint8_t_from_tlv_data(data, &tlv_extracted->structure_type);
-}
-
-static bool handle_version(const tlv_data_t *data, tlv_extracted_t *tlv_extracted) {
-    CX_ASSERT(cx_hash_update((cx_hash_t *) &tlv_extracted->hash_ctx, data->raw, data->raw_size));
-    return get_uint8_t_from_tlv_data(data, &tlv_extracted->version);
-}
-
-static bool handle_coin_type(const tlv_data_t *data, tlv_extracted_t *tlv_extracted) {
-    CX_ASSERT(cx_hash_update((cx_hash_t *) &tlv_extracted->hash_ctx, data->raw, data->raw_size));
-    return get_uint32_t_from_tlv_data(data, &tlv_extracted->coin_type);
-}
-
-static bool handle_application_name(const tlv_data_t *data, tlv_extracted_t *tlv_extracted) {
-    CX_ASSERT(cx_hash_update((cx_hash_t *) &tlv_extracted->hash_ctx, data->raw, data->raw_size));
-    return get_string_from_tlv_data(data,
-                                    tlv_extracted->application_name,
-                                    1,
-                                    sizeof(tlv_extracted->application_name));
-}
-
-static bool handle_ticker(const tlv_data_t *data, tlv_extracted_t *tlv_extracted) {
-    CX_ASSERT(cx_hash_update((cx_hash_t *) &tlv_extracted->hash_ctx, data->raw, data->raw_size));
-    return get_string_from_tlv_data(data, tlv_extracted->ticker, 1, sizeof(tlv_extracted->ticker));
-}
-
-static bool handle_magnitude(const tlv_data_t *data, tlv_extracted_t *tlv_extracted) {
-    CX_ASSERT(cx_hash_update((cx_hash_t *) &tlv_extracted->hash_ctx, data->raw, data->raw_size));
-    return get_uint8_t_from_tlv_data(data, &tlv_extracted->magnitude);
-}
+DEFINE_TLV_PARSER(TUID_TLV_TAGS, NULL, parse_dynamic_token_tuid)
```

Comment:
Shouldn't we put this part into `tlv_use_case_dynamic_descriptor()`function ? I mean, once the `parse_dynamic_token_tag()` is called, we could call `parse_dynamic_token_tuid()` to parse the TUID buffer. In this case, we wouldn't need anymore to call `DEFINE_TLV_PARSER` macro in the app.

Reply:
the TUID buffer is application specific, the library can't know the internal tags and types

Reply:
Ok understood